### PR TITLE
#49 Feature/event ticket price

### PR DESCRIPTION
**Pull request by human453453467**

### CODE REVIEW

File: src/main/java/by/ladyka/club/service/ClubEventTicketPriceServiceImpl.java [hunk 1]
```diff
@@ -1,41 +1,139 @@
...
+            throw new AccessControlException("This has't access of this event");
+        }
+    }
+
+    @Override
+    public ClubEventTicketPriceDTO save(ClubEventTicketPriceDTO requestDto, String username){
+        UserEntity user = userService.getUserEntity(username);
+        EventEntity event = eventsService.getEventById(requestDto.getEventId()).orElseThrow(() -> new RuntimeException("impossible to create a price for a non-existent event"));
+        ClubEventTicketPrice clubEventTicketPrice;
+        if(eventsService.hasAccess(event, user)){
```
Comment (ladyka) [inline src/main/java/by/ladyka/club/service/ClubEventTicketPriceServiceImpl.java]: ONE if

File: src/main/java/by/ladyka/club/service/ClubEventTicketPriceServiceImpl.java [hunk 2]
```diff
@@ -1,41 +1,139 @@
...
+            if (requestDto.getId() == null || requestDto.getId() < 1) {
```
Comment (ladyka) [inline src/main/java/by/ladyka/club/service/ClubEventTicketPriceServiceImpl.java]: Second

File: src/main/java/by/ladyka/club/service/ClubEventTicketPriceServiceImpl.java [hunk 3]
```diff
@@ -1,41 +1,139 @@
...
+                clubEventTicketPrice = new ClubEventTicketPrice();
+                clubEventTicketPrice = converterClubEventTicketPriceService.toEntity(requestDto, clubEventTicketPrice, null);
+            }else {
+                Optional<ClubEventTicketPrice> clubEventTicketPriceOptional = clubEventTicketPriceRepository.findById(requestDto.getId());
+                if(clubEventTicketPriceOptional.isPresent()){
```
Comment (ladyka) [inline src/main/java/by/ladyka/club/service/ClubEventTicketPriceServiceImpl.java]: Third

File: src/main/java/by/ladyka/club/service/ClubEventTicketPriceServiceImpl.java [hunk 4]
```diff
@@ -1,41 +1,139 @@
...
+                    clubEventTicketPrice = clubEventTicketPriceOptional.get();
+                    if(orderEntityRepository.countByClubEventTicketPriceDanceOrClubEventTicketPriceTable(clubEventTicketPrice, clubEventTicketPrice) > 0){
```
Comment (ladyka) [inline src/main/java/by/ladyka/club/service/ClubEventTicketPriceServiceImpl.java]: 4

File: src/main/java/by/ladyka/club/service/ClubEventTicketPriceServiceImpl.java [hunk 5]
```diff
@@ -1,41 +1,139 @@
...
+
+    @Override
+    public Optional<ClubEventTicketPrice> getLowPriceEntityForEventTablePlace(EventEntity event) {
+        return getLowPriceForEventByPriceType(event.getId(), EventTicketPriceType.table);
     }
 
 	@Override
 	public BigDecimal getLowPriceForEventDance(EventEntity event) {
-		return getLowPriceForEventByPriceType(event.getId(), EventTicketPriceType.dance).orElse(BigDecimal.valueOf(0L));
+        Optional<ClubEventTicketPrice> clubEventTicketPriceOptional = getLowPriceForEventByPriceType(event.getId(), EventTicketPriceType.dance);
```
Comment (ladyka) [inline src/main/java/by/ladyka/club/service/ClubEventTicketPriceServiceImpl.java]: Why?

File: src/main/java/by/ladyka/club/service/ClubEventTicketPriceServiceImpl.java [hunk 6]
```diff
@@ -1,41 +1,139 @@
...
+		    return clubEventTicketPriceOptional.get().getCost();
+        }else{
+		    return BigDecimal.valueOf(0L);
+        }
 	}
 
 	@Override
 	public BigDecimal getLowPriceForEventTablePlace(EventEntity event) {
-		return getLowPriceForEventByPriceType(event.getId(), EventTicketPriceType.table).orElse(BigDecimal.valueOf(0L));
+        Optional<ClubEventTicketPrice> clubEventTicketPriceOptional = getLowPriceForEventByPriceType(event.getId(), EventTicketPriceType.table);
```
Comment (ladyka) [inline src/main/java/by/ladyka/club/service/ClubEventTicketPriceServiceImpl.java]: Why?

File: src/main/java/by/ladyka/club/repository/ClubEventTicketPriceRepository.java [hunk 7]
```diff
@@ -1,17 +1,49 @@
...
 @Repository
 public interface ClubEventTicketPriceRepository extends JpaRepository<ClubEventTicketPrice, Long> {
     @Query("Select p from ClubEventTicketPrice p inner join p.event e where e.id = :eventId AND p.type = :eventTicketPriceType order by p.cost ASC")
     List<ClubEventTicketPrice> findAscSortPricesForEventByPriceType(@Param("eventId") Long eventId, @Param("eventTicketPriceType") EventTicketPriceType eventTicketPriceType, Pageable pageable);
+
+    @Query("SELECT p " +
+            "FROM ClubEventTicketPrice p " +
+            "LEFT JOIN p.danceOrders o " +
+            "WHERE :now BETWEEN p.startActiveTime AND p.endActiveTime " +
+            "AND p.type = 'dance' " +
```
[the remark this answers is not in the document]
Comment (human453453467) [reply src/main/java/by/ladyka/club/repository/ClubEventTicketPriceRepository.java]: as i know it doesn't possible in java annotation

File: src/main/java/by/ladyka/club/repository/ClubEventTicketPriceRepository.java [hunk 8]
```diff
@@ -1,17 +1,49 @@
...
+
+
+
+
+    @Query(value = "SELECT p " +
+            "FROM ClubEventTicketPrice p " +
+            "LEFT JOIN p.tableOrders o " +
+            "LEFT JOIN o.tableNumbers i " +
+            "WHERE :now BETWEEN p.startActiveTime AND p.endActiveTime " +
+            "AND p.type = 'table' " +
```
[the remark this answers is not in the document]
Comment (human453453467) [reply src/main/java/by/ladyka/club/repository/ClubEventTicketPriceRepository.java]: as i know it doesn't possible in java annotation

**Pull request closed without merging**